network/networkutils/networkutils_linux.go

```diff
@@ -58,15 +59,21 @@ func NewNetworkUtils(nl netlink.NetlinkInterface, plClient platform.ExecClient)
 func (nu NetworkUtils) CreateEndpoint(hostVethName, containerVethName string) error {
 	log.Printf("[net] Creating veth pair %v %v.", hostVethName, containerVethName)
 
+	mac, err := net.ParseMAC(defaultHostVethHwAddr)
+	if err != nil {
+		log.Printf("[net] Failed to parse the mac addrress %v", defaultHostVethHwAddr)
+		return newErrorNetworkUtils(err.Error())
+	}
 	link := netlink.VEthLink{
 		LinkInfo: netlink.LinkInfo{
-			Type: netlink.LINK_TYPE_VETH,
-			Name: hostVethName,
+			Type:       netlink.LINK_TYPE_VETH,
+			Name:       hostVethName,
+			MacAddress: mac,
```

Comment:
lets pass this macaddress as parameter? lets do this only for singletenancy transparent mode scenario..not want to change this to other scenarios.

Reply:
Sure